examples/3d/generate_custom_mesh.rs

```diff
@@ -0,0 +1,238 @@
+// ! This example demonstrates how to create a custom mesh,
+// ! assign a custom UV mapping for a custom texture,
+// ! and how to change the UV mapping at run-time.
+use bevy::prelude::*;
+use bevy::render::mesh::Indices;
+use bevy::render::mesh::VertexAttributeValues;
+use bevy::render::render_resource::PrimitiveTopology;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, input_handler)
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: ResMut<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+) {
+    // Importing the custom texture.
+    let custom_texture_handle: Handle<Image> = asset_server.load("textures/array_texture.png");
+    // Creating and saving a handle to the mesh.
+    let cube_mesh_handle: Handle<Mesh> = meshes.add(create_cube_mesh());
+
+    // Rendering the mesh with the custom texture using a PbrBundle.
+    commands.spawn(PbrBundle {
+        mesh: cube_mesh_handle,
+        material: materials.add(StandardMaterial {
+            base_color_texture: Some(custom_texture_handle),
+            ..default()
+        }),
+        ..default()
+    });
+
+    // Transform for the camera and lighting, looking at (0,0,0) (the position of the mesh).
+    let camera_and_light_transform =
+        Transform::from_xyz(2.1, 2.1, 2.1).looking_at(Vec3::ZERO, Vec3::Y);
+
+    // Camera in 3D space.
+    commands.spawn(Camera3dBundle {
+        transform: camera_and_light_transform,
+        ..default()
+    });
+
+    // Lighting up the scene.
+    commands.spawn(PointLightBundle {
+        point_light: PointLight {
+            intensity: 1000.0,
+            range: 100.0,
+            ..default()
+        },
+        transform: camera_and_light_transform,
+        ..default()
+    });
+}
+
+// System to recieve input from the user, check out examples/input/ for more examples about user
+// input.
+fn input_handler(
+    keyboard_input: Res<Input<KeyCode>>,
+    mesh_query: Query<&Handle<Mesh>>,
```

Comment:
I would have preferred if the mesh had a marker component and querying for it used the marker component `Query<&Handle<Mesh>, With<CustomUv>>`. Since that would be how you would do it in a real word use-case.

But I think its not necessary. So up to you to add it or not.